### ISSUE TITLE
Add optional values argument to Query constructor

### ISSUE THREAD

**Issue by joeconwaystk** [stablekernel/aqueduct#459]
For queries that will insert/update rows, an optional argument with those values is a useful shortcut that can infer the `T` of `Query<T>`. 

Usage:

```dart
final thing = new Thing(...);
final q = new Query(context, values: thing)
  ..where((t) => t.id).equalTo(1); // is inferred to Query<Thing>
await q.update();
```

**Comment by jhdoak** [comment]
@joeconwaystk - Can this issue be closed with the merge of #558?

**Comment by joeconwaystk** [comment]
Oh yup, thank you!
